asm/f32/dot.go

```diff
@@ -4,18 +4,18 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-package asm
+package f32
 
-func DsdotUnitary(x, y []float32) (sum float64) {
+func DotUnitary(x, y []float32) (sum float32) {
 	for i, v := range x {
-		sum += float64(y[i]) * float64(v)
+		sum += y[i] * v
 	}
 	return
 }
 
-func DsdotInc(x, y []float32, n, incX, incY, ix, iy uintptr) (sum float64) {
```

Comment:
Explanation in [Related Comment](https://github.com/gonum/internal/pull/30/commits/55f5434be290a35b149ff3e234bd3c7999ee3784#r64160952)

I mucked up the file movement and dsdot/sdot crossed paths.